### PR TITLE
Fix: Ignore non em services

### PR DESCRIPTION
**Pull request by duncanhall**
Service discovery results from Consul can include infrastructure services that were not installed via Environment Manager - we should not include these in any service lists. See the last service in the image below for an example:

selection_028

This fix filters service discovery results to only include services that have a `Tags` object containing a `deployment_id` field.

### CODE REVIEW

File: server/modules/service-discovery/consul/consulCatalog.js [hunk 1]
```diff
@@ -39,7 +39,17 @@ function getAllNodes(environment) {
 
 function getNode(environment, nodeName) {
   assert(nodeName, 'nodeName is required');
-  return executeConsul(environment, clientInstance => clientInstance.catalog.node.services(nodeName));
+  return executeConsul(environment, clientInstance => clientInstance.catalog.node.services(nodeName))
+    .then((nodes) => {
+      if (!nodes || !nodes.Services) {
```
Comment (Merlin-Taylor) [inline server/modules/service-discovery/consul/consulCatalog.js]: In the case where `nodes.Services` is falsey, should we return `null` or `nodes` with `nodes.Services = []`?

Comment (duncanhall) [inline server/modules/service-discovery/consul/consulCatalog.js]: Yep, good catch, have updated accordingly.

Comment (filso) [inline server/modules/service-discovery/consul/consulCatalog.js]: maybe we could use strict comparisons here?

Comment (duncanhall) [inline server/modules/service-discovery/consul/consulCatalog.js]: Not sure it's necessary. These are properties on an object returned from the consul SDK, so there's no ambiguity.

**Pull request merged**